editor/import/3d/scene_import_settings.cpp

```diff
@@ -1448,7 +1448,8 @@ void SceneImportSettingsDialog::_save_dir_callback(const String &p_path) {
 					item->set_metadata(0, E.key);
 					item->set_editable(0, true);
 					item->set_checked(0, true);
-					String path = p_path.path_join(name);
+					// Replacing | (pipe) with _ (underscore) for compatibility with the Windows file system.
+					String path = p_path.path_join(name.replace("|", "_"));
```

Comment:
I would consider using `String::validate_filename()` instead of specifically replacing `|` with `_`. Basically, there are many illegal filename characters and we should replace all of them.

the validate_filename will replace characters with _ so it should work the same for the case you did.

Reply:
Thank you for your replies.

As far as I know, this only automatically happens in animation names in Blender, but I guess anyone could actually insert | in whatever they decide to name their material and meshes names.

I agree `String::validate_filename()` is better, I didn't know it existed and it's perfect for the use case. I guess I'll change that and wrap both the material and mesh names in the method as well.

A cool addition to this imo would be to display an error message or a pop-up whenever a user is trying to save their resources using invalid characters, as the previous error was a little cryptic. I don't know how much value there is in letting non-windows users use those characters in filenames either. What do you think would be the most appropriate solution?

For now, I'm gonna make the quick changes you mentioned and force-push them asap!